torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -597,6 +774,232 @@ static void PrepareForRemoveMutations(MutationRemover& mr, Block* b) {
+      auto attrModule = module_;
+
+      auto moduleNames =
+          findSubModuleAttr(n->inputs().at(0), name, attrModule, graph);
+      if (!attrModule.hasattr(name))
+        continue;
+      auto attr = attrModule.attr(name);
+      Value* paramConst = nullptr;
+
+      std::string fullName("");
+      for (auto& name : moduleNames) {
+        fullName += name + '.';
+      }
+      fullName += name;
+
+      auto type = attrModule.type();
+      auto slot = *type->findAttributeSlot(name);
+
+      // Add model_parameters and model_buffers as model inputs. Order is
+      // preserved based on the appearance in the graph.
+      if (type->is_parameter(slot) || type->is_buffer(slot)) {
+        if (attrValues.find(fullName) != attrValues.end()) {
+          paramConst = attrValues[fullName];
+        } else {
+          paramConst = findArgumentAsInputParam(graph, fullName, attr);
+          if (!paramConst) {
+            toDestory.emplace_back(n);
+            continue;
+          }
+          attrValues.insert({fullName, paramConst});
+        }
+      }
+
+      if (n->kind() == prim::SetAttr) { // Handle SetAttr node
+        if (attrModule.hasattr(name)) {
+          // If inside a block, keep the output value to register in block
+          // output.
+          auto block_ = n->owningBlock();
+          if (block_->owningNode() &&
+              block_->owningNode()->kind() == prim::If) { // TODO: Add loop
+
+            auto attrValue =
+                (setAttrValues.find(fullName) != setAttrValues.end())
+                ? setAttrValues[fullName]
+                : attrValues[fullName];
+
+            auto blockOutput = registerSetAttrInBlocks(
+                graph, block_, n->inputs().at(1), attrValue, fullName);
+            nextSetAttrValues[fullName] = blockOutput;
+          }
+          // SetAttr writes a value to an attr. Keep this
+          // in the setAttrValues map.
+          setAttrValues[fullName] = n->inputs().at(1);
+          toDestory.emplace_back(n);
+        }
+      } else if (n->kind() == prim::GetAttr) { // Handle GetAttr node
+        if (setAttrValues.find(fullName) != setAttrValues.end()) {
+          // Attr has been set earlier in the graph.
+          // Read its value from setAttrValues map.
+          auto set_attr_node_input = setAttrValues[fullName];
+          // Clone SetAttr input
+          auto cloneNode = insertCloneBeforeNode(graph, set_attr_node_input, n);
+          n->output()->replaceAllUsesAfterNodeWith(n, cloneNode->output());
+          toDestory.emplace_back(n);
+        } else if (attrValues.find(fullName) != attrValues.end()) {
+          // Attr has not been set earlier in the graph. Replace it with the
+          // graph parameter if exists.
+          n->output()->replaceAllUsesWith(attrValues[fullName]);
+          n->removeAllInputs();
+          toDestory.emplace_back(n);
+        }
+      }
+    } else if (
+        mr.inplaceOpVariant(n) &&
```

Comment:
this is only usage of `mr` right? can `mr` be passed as const reference?

Reply:
Making this const would not work, since the inplaceOpVariant() method is not declared as const.